<?xml version="1.0" ?>
<sdf version="1.6">
  <world name="denso_cubes_task">
    <gravity>0 0 -9.81</gravity>
    <physics name="default_physics" default="0" type="ode">
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>1</shadows>
    </scene>
    <!-- Light Source -->
    <include>
      <uri>model://sun</uri>
    </include>
    <!-- A ground plane -->
    <include>
      <uri>model://ground_plane</uri>
      <pose>0 0 0 0 0 0</pose>
    </include>
    <!-- The robot base -->
    <model name="denso_base">
      <include>
        <uri>model://denso_base</uri>
        <pose>0 0 0 0 0 0</pose>
      </include>
    </model>
    <!-- Table -->
    <!-- Table top is at (x,y,z) = (0.7, 0, 0.8) -->
    <!-- Table top size 0.9 x 0.9 -->
    <model name="cafe_table">
      <include>
        <uri>model://cafe_table</uri>
        <pose>0.7 0 0 0 0 0</pose>
      </include>
    </model>
    <!-- Cubes -->
    <model name="cube01">
      <include>
        <uri>model://wood_cube_5cm</uri>
        <pose>0.6 -0.2 0.775 0 0 0</pose>
      </include>
    </model>
    <model name="cube02">
      <include>
        <uri>model://wood_cube_5cm</uri>
        <pose>0.6 0 0.775 0 0 0</pose>
      </include>
    </model>
    <model name="cube03">
      <include>
        <uri>model://wood_cube_5cm</uri>
        <pose>0.6 0.2 0.775 0 0 0</pose>
      </include>
    </model>
  </world>
</sdf>
